dnsconfig.js

```diff
@@ -19,6 +19,96 @@ function loadTemplate(template) {
   };
 }
 
+/**
+ * Create A, optional AAAA, and optional HTTPS (SVCB) records for an OSM web service,
+ * with optional Cloudflare proxy support.
+ *
+ * @param {string} name                - Hostname (e.g. "www").
+ * @param {string[]} ipv4s             - Required array of IPv4 addresses.
+ * @param {string[]} [ipv6s=[]]        - Optional array of IPv6 addresses (defaults to empty).
+ * @param {boolean} [enableHttps=true] - Whether to create an HTTPS record (default: true).
+ * @param {boolean} [cloudflareProxy=false] - Whether to enable Cloudflare proxy (default: false).
+ *
+ */
+function osm_web_service(
+  name,
+  ipv4s,
+  ipv6s,
+  enableHttps,
+  cloudflareProxy
+) {
+
+  // If ipv6s was not provided, default to empty array
+  if (ipv6s === undefined) {
```

Comment:
This is to workaround the dnscontrol javascript engine not supporting setting function parameter defaults.